lib/rendition.js

```diff
@@ -42,17 +102,158 @@ class Rendition {
     }
 
     size() {
-        return this.metadata[RenditionMetadata.REPO_SIZE];
+        return fs.statSync(this.path).size;
+    }
+
+    async sha1() {
+        if (!this._sha1) {
+            this._sha1 = await fileHash(this.path, HASH_ALGORITHM);
+        }
+        return this._sha1;
     }
 
-    sha1() {
-        return this.metadata[RenditionMetadata.REPO_SHA1];
+    exists() { // see and read the file
+        return fs.existsSync(this.path);
+    }
+
+    async setMimeInformation(mime=null, encoding=null){
+        // for javascript workers mostly ------------------------------------------------------
+        // explicitly set mimetype would get priority
+        if(mime){ // to handle case when this._mime is set to default DEFAULT_CONTENT_TYPE
+            this._mime = mime;
+        }
+
+        // explicitly set encoding would get priority
+        if(encoding){ // to handle case when this.encoding would have a default value
+            this._encoding = encoding;
+        }
+
+        // shellscript mostly ------------------------------------------------------
+        if(this.mimeInfoPath && await fs.pathExists(this.mimeInfoPath)){ // use the path containing the file with mime information
```

Comment:
This logic should be in the shellscript.js (see comment over there).

Reply:
Double-check:

You mean after the processing (line https://github.com/adobe/asset-compute-sdk/blob/e866028b9d3638fa9caad455037cef8bbae0b379/lib/shell/shellscript.js#L55), right?

So the shellscript still can write its file and there is no usage change on that end.

Reply:
yes, after the shell script was invoked in `processWithScript()` it checks if the type file exists and parses it, and calls rendition.setContentType(). just like the error file is also handled in there (inside newScriptError()).